demos/projects/ESPRESSIF/esp32/main/azure_iot_freertos_esp32_main.c

```diff
@@ -1,274 +1,263 @@
+
+    /* Reduce ulGlobalEntryTimeMs from obtained time so as to always return the
+     * elapsed time in the application. */
+    ulTime = ( uint64_t ) ( ulTime + ulGlobalEntryTime );
+
+    return ulTime;
+}
+
+/*-----------------------------------------------------------*/
+
+static xSemaphoreHandle s_semph_get_ip_addrs;
+static esp_ip4_addr_t s_ip_addr;
+
+#define NR_OF_IP_ADDRESSES_TO_WAIT_FOR 1
+
+#if CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_FAST
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_FAST_SCAN
+#elif CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_ALL_CHANNEL
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_ALL_CHANNEL_SCAN
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SIGNAL
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SIGNAL
+#elif CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SECURITY
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SECURITY
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_AUTH_OPEN
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_OPEN
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WEP
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WEP
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_ENTERPRISE
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_ENTERPRISE
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WAPI_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WAPI_PSK
+#endif
+
+/**
+ * @brief Checks the netif description if it contains specified prefix.
+ * All netifs created withing common connect component are prefixed with the module TAG,
+ * so it returns true if the specified netif is owned by this module
+ */
+static bool is_our_netif(const char *prefix, esp_netif_t *netif)
+{
+    return strncmp(prefix, esp_netif_get_desc(netif), strlen(prefix) - 1) == 0;
+}
+
+static void on_got_ip(void *arg, esp_event_base_t event_base,
+                      int32_t event_id, void *event_data)
+{
+    ip_event_got_ip_t *event = (ip_event_got_ip_t *)event_data;
+    if (!is_our_netif(TAG, event->esp_netif)) {
+        ESP_LOGW(TAG, "Got IPv4 from another interface \"%s\": ignored", esp_netif_get_desc(event->esp_netif));
+        return;
+    }
+    ESP_LOGI(TAG, "Got IPv4 event: Interface \"%s\" address: " IPSTR, esp_netif_get_desc(event->esp_netif), IP2STR(&event->ip_info.ip));
+    memcpy(&s_ip_addr, &event->ip_info.ip, sizeof(s_ip_addr));
+    xSemaphoreGive(s_semph_get_ip_addrs);

```

Comment:
sorry for the nit here (and elsewhere in this file). I know this is pasted from somewhere else but any chance we could get the same spacing between variables and if/else/while blocks?
```suggestion
    ip_event_got_ip_t *event = (ip_event_got_ip_t *)event_data;
    
    if (!is_our_netif(TAG, event->esp_netif)) {
        ESP_LOGW(TAG, "Got IPv4 from another interface \"%s\": ignored", esp_netif_get_desc(event->esp_netif));
        return;
    }
    
    ESP_LOGI(TAG, "Got IPv4 event: Interface \"%s\" address: " IPSTR, esp_netif_get_desc(event->esp_netif), IP2STR(&event->ip_info.ip));
    memcpy(&s_ip_addr, &event->ip_info.ip, sizeof(s_ip_addr));
    xSemaphoreGive(s_semph_get_ip_addrs);

```

Reply:
No problem proposing the change, but as I mentioned to Himanshu, I want to keep these parts as is so it gets easier for Espressif to later port this sample to esp-idf.